### PR TITLE
Allow resurrected units to act on the current turn if they have not yet had their turn

### LINKED ISSUE #6783: Combat, Resurrect behavior (version 1.0.1 6629)
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In the original game, when a decimated troop is resurrected, it must play in the current turn.
In fheroes2, this is not the case.

This must also be the case with the Resurrect True and Animate Dead spells.

In original game:

https://user-images.githubusercontent.com/43583994/223733519-23b80271-abea-4052-af95-f122a2317253.mp4

In fheroes2:

https://user-images.githubusercontent.com/43583994/223733612-4ba12b13-a01f-499b-83e0-894c08181775.mp4

### Save file

Here are the save files:
[Resurrect behavior 01.zip](https://github.com/ihhub/fheroes2/files/10921284/Resurrect.behavior.01.zip)

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #6783

### CODE REVIEW

File: src/fheroes2/battle/battle_troop.cpp [hunk 1]
```diff
@@ -1528,16 +1520,17 @@ void Battle::Unit::SpellRestoreAction( const Spell & spell, uint32_t spoint, con
     case Spell::RESURRECT:
     case Spell::ANIMATEDEAD:
     case Spell::RESURRECTTRUE: {
-        // remove from graveyard
         if ( !isValid() ) {
-            // TODO: buggy behaviour
```
Comment (oleg-derevenetz) [inline src/fheroes2/battle/battle_troop.cpp:1533]: This comment is from #440, I have no idea what does it mean, but most likely there is no buggy behavior anymore.

**Comment by oleg-derevenetz** [comment]
Hi @LeHerosInconnu could you please verify this PR in different situations?

**Comment by oleg-derevenetz** [comment]
@Branikolog I didn't encounter this during testing. Anyway, let's implement the "natural" case, and if later there will be any special case with clear logic, this can be implemented separately.

**Comment by Branikolog** [comment]
@oleg-derevenetz 

> let's implement the "natural" case,

Sure!
I just started to test your PR comparing it with the OG behaviour, where, as I remember, creatures could miss their turn after being resurrected.
I thought after dead creature missed its expected turn (when creatures with lower speed start acting) and then you resurrect this creature (within the current round), it shouldn't perform action till the next round begins. But somehow in the OG game it sometimes skip its turn after being resurrected but sometimes it's able to act. All other cases resurrected creature worked as expected. 
Anyway, your PR works well for me.

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , everything looks great. I left only one comment to verify the logic with the OG. Could you please take a look?

File: src/fheroes2/battle/battle_troop.cpp [hunk 2]
```diff
@@ -664,22 +665,16 @@ void Battle::Unit::PostKilledAction()
...
         ResetModes( CAP_MIRROROWNER );
     }
+
     // Remove mirror image (slave)
     if ( Modes( CAP_MIRRORIMAGE ) && mirror != nullptr ) {
         mirror->ResetModes( CAP_MIRROROWNER );
         mirror = nullptr;
     }
 
-    ResetModes( TR_RESPONDED );
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/battle/battle_troop.cpp:673]: No, it does not. Crusaders here do not retaliate twice:

https://user-images.githubusercontent.com/32623900/224470349-996b5e87-d7b5-4594-97f1-bbd2b559a196.mp4

**Review by ihhub — APPROVED** [review]

**Comment by LeHerosInconnu** [comment]
Hello @oleg-derevenetz,

> Hi @LeHerosInconnu could you please verify this PR in different situations?

I did some tests, also with the slow spell, it seems to work fine.

**Review by Branikolog — APPROVED** [review]
Works nice!

**Comment by ihhub** [comment]
@oleg-derevenetz , huge thanks for this improvement!

**Pull request merged**